Automattic/sensei
# Add shared settings for module borders and styles

### Changes proposed in this Pull Request

* Removes Borderless and Bordered styles from course outline block and turns it into a border toggle.
* Adds Filled and Minimal styles to the course outline block. 
* Adds a Border toggle and a Border color setting in the module block.
* All new settings will override previously set values. So for example when a new setting is set on the outline level, all modules will get that setting regardless if a different values was set in the module level previously.
* The outline settings are not sychronised with the modules which means that if all modules are set to a different setting individually, the outline module setting will not get updated to reflect that. This was done to avoid extra complexity.
* The 'Styles' setting is provided entirely by gutenberg and cannot be customised. For this reason I ended up defining `applyParentStyle` to monitor changes on the outline styles element.
* _Edit:_ Fixed also an issue with the lesson count on the progress bar were the lessons from the examples and previews were included in the total count.

### Testing instructions

* Create a new outline block.
* Try setting values for the style in both the outline block and in individual levels. Observe that the changes are reflected in the frontend.
* Try setting different values for the Border toggle in both the outline and module level. Observe that the changes are reflected in the frontend.
* Set different color values for the border and observe the changes.
* Check that setting a border color automatically enables the border. Also clearing the color disables it.
